django/db/models/query_utils.py

```diff
@@ -43,14 +43,11 @@ def _combine(self, other, conn):
         if not(isinstance(other, Q) or getattr(other, 'conditional', False) is True):
             raise TypeError(other)
 
-        # If the other Q() is empty, ignore it and just use `self`.
-        if not other:
+        if not self:
+            return other
```

Comment:
Still have to figure out if we want to return `copy.copy` here, I guess it wouldn't hurt. In all cases we can't assume that `other` has a `deconstruct` method like we previously did.

Reply:
Agreed we should return a copy. I pushed edits with extra tests.